File: Server/Server-Side/TeamApp/TeamApp.Infrastructure.Persistence/Entities/Message.cs
using System;
using System.Collections.Generic;

namespace TeamApp.Infrastructure.Persistence.Entities
{
    public partial class Message
    {
        public string MessageId { get; set; }
        public string MessageUserId { get; set; }
        public string MessageGroupChatId { get; set; }
        public string MessageContent { get; set; }
        public DateTime? MessageCreatedAt { get; set; }
       
        public virtual GroupChat MessageGroupChat { get; set; }
        public virtual User MessageUser { get; set; }
    }
}

File: Client/chat_app/lib/models/message.dart
class Message {
  String messageId;
  String messageUserId;
  String messageGroupChatId;
  String messageContent;
  DateTime messageCreatedAt;
  String messengerUserAvatar;
  String messengerUserName;

  Message(
      {required this.messageId,
      required this.messageUserId,
      required this.messageGroupChatId,
      required this.messageContent,
      required this.messageCreatedAt,
      required this.messengerUserAvatar,
      required this.messengerUserName});

  factory Message.fromJson(Map<String, dynamic> json) => Message(
      messageId: json['messageId'],
      messageUserId: json['messageUserId'],
      messageGroupChatId: json['messageGroupChatId'],
      messageContent: json['messageContent'],
      messageCreatedAt: DateTime.parse(json['messageCreatedAt']),
      messengerUserAvatar: json['messengerUserAvatar'],
      messengerUserName: json['messengerUserName']);
}
